### PR TITLE
Switch to using webpush taxonomy type for notify setter

### PR DESCRIPTION
**Pull request by aosmond**

### CODE REVIEW

**Comment by aosmond** [comment]
@Yoric r?

File: src/adapters/webpush/mod.rs [hunk 1]
```diff
@@ -222,6 +216,16 @@ impl<C: Controller> Adapter for WebPush<C> {
                 NO_AUTH_USER_ID
             };
 
+            if id == self.setter_notify_id {
+                return match value {
+                    Value::WebPushNotify(ref v) => {
+                        self.set_notify(v).unwrap();
+                        (id, Ok(()))
+                    },
+                    _ => (id, Err(Error::InternalError(InternalError::InvalidInitialService)))
```
Comment (Yoric) [inline src/adapters/webpush/mod.rs]: Actually, that's an `Error::TypeError`.

**Comment by Yoric** [comment]
r=me with a small change

**Comment by ferjm** [comment]
@aosmond can we close this one now that #380 landed?

**Pull request closed without merging**